csrc/fusion_segmenter.h

```diff
@@ -501,7 +502,8 @@ struct SegmentCandidateFinderOptions {
   bool run_combine_reductions = true;
   bool run_herrmann_merge = true;
   bool run_final_merge = true;
-  bool only_segment_resharding_exprs = false;
+  std::function<bool(SegmentedGroup*, SegmentedGroup*)>
```

Comment:
I understand this is convenient and am not against adding it, but please add detailed comments how this extension point can be used.

Reply:
I added a comment in the last commit, lmk

Reply:
Thanks! Looks good to me.